apps/payments/next/app/[locale]/[offeringId]/[interval]/new/route.ts

```diff
@@ -2,36 +2,55 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import { setupCartAction } from '@fxa/payments/ui/actions';
 import { auth } from 'apps/payments/next/auth';
 import { headers } from 'next/headers';
 import { redirect } from 'next/navigation';
+import { NextRequest } from 'next/server';
+import { setupCartAction } from '@fxa/payments/ui/actions';
+import { CartEligibilityStatus } from '@fxa/shared/db/mysql/account';
+
+export const dynamic = 'force-dynamic';
 
 interface CheckoutNewParams {
   locale: string;
   offeringId: string;
   interval: string;
 }
 
-export default async function CheckoutNew({
-  params,
-  searchParams,
-}: {
-  params: CheckoutNewParams;
-  searchParams: Record<string, string>;
-}) {
+export async function GET(
+  request: NextRequest,
+  {
+    params,
+  }: {
+    params: CheckoutNewParams;
+  }
+) {
   const { offeringId, interval } = params;
-  const { coupon } = searchParams;
+  const redirectToUrl = request.nextUrl.clone();
+  const searchParams = request.nextUrl.searchParams;
+  const coupon = searchParams.get('coupon') || undefined;
   const session = await auth();
 
   const fxaUid = session?.user?.id;
   const ip = (headers().get('x-forwarded-for') ?? '127.0.0.1').split(',')[0];
 
+  let eligibilityStatus = 'checkout';
```

Comment:
nit: this can be deleted, and the variable can be declared as a const down on line 50

Reply:
Per discussion, this change cannot be made as `eligibilityStatus` would not be able to be utilized if declared in the try/catch.